src/c++/perf_analyzer/test_report_writer.cc

```diff
@@ -90,4 +103,50 @@ TEST_CASE("testing WriteGpuMetrics")
   }
 }
 
+TEST_CASE("report_writer: WriteLlmMetrics")
+{
+  // Create a dummy request records
+  using std::chrono::system_clock;
+  using std::chrono::time_point;
+
+  Experiment experiment;
+  auto clock_epoch{time_point<system_clock>()};
+
+  uint64_t seq_id1{123};
+  auto request1{clock_epoch + std::chrono::microseconds(1)};
+  auto response1{clock_epoch + std::chrono::microseconds(4)};
+  auto response2{clock_epoch + std::chrono::microseconds(5)};
+
+  RequestRecord rr1{
+      request1, std::vector<time_point<system_clock>>{response1, response2},
+      0,        false,
+      seq_id1,  false};
+
+  uint64_t seq_id2{456};
+  auto request2{clock_epoch + std::chrono::microseconds(4)};
+  auto response3{clock_epoch + std::chrono::microseconds(5)};
+  auto response4{clock_epoch + std::chrono::microseconds(7)};
+
+  RequestRecord rr2{
+      request2, std::vector<time_point<system_clock>>{response3, response4},
+      0,        false,
+      seq_id2,  false};
+
+  std::vector<RequestRecord> request_records{rr1, rr2};
+  experiment.requests = std::move(request_records);
+  std::vector<Experiment> experiments{experiment};
+
+  // Avg first token latency
+  // = ((response1 - request1) + (response3 - request2)) / 2
+  // = (3 + 1) / 2 = 2 us
+  // Avg token-to-token latency
+  // = ((response2 - response1) + (response4 - response3)) / 2
```

Comment:
let's have at least one case where a request receives multiple responses

Reply:
This case has two requests with two responses each: 
- `request1`: `response2`, `response3`
- `request2`: `response3`, `response4`

Are you saying we should have a case where requests have more responses (>2)?

Reply:
Sorry, I misunderstood the code/comments. Regardless, I'd like to see cases of 1 response, 2 responses, and 3 or more responses. And a case of 0 responses if that is legal.

Reply:
I think we would need to decide what gets put into the csv for the case where there is one response (t2t latency is undefined) and for the case where there is zero responses (ttft and t2t are undefined). Perhaps "N/A" is an acceptable value in those cases?

That would likely require changing [this](https://github.com/triton-inference-server/client/pull/463/files#diff-20e536474a07e95bac763da8a8aedcbfe08bd4a30eb5d405c85ad8a13a07dc15R100) function to return possibly [std::optional](https://en.cppreference.com/w/cpp/utility/optional) of whatever values are currently returned. Cool usage of new C++17 feature now that we support it.

Reply:
Does other server/client side stats have default value for undefined numbers? If not, I'm fine setting it to "n/a".

Reply:
Changed to output n/a when values are not defined.

Reply:
Correct me if I'm wrong, but I don't think this has been resolved:
_I'd like to see cases of 1 response, 2 responses, and 3 or more responses. And a case of 0 responses if that is legal._

Reply:
> I'd like to see cases of 1 response, 2 responses, and 3 or more responses.

I added cases for zero, single, and >3 responses. Anything greater than single response should behave the same. 

> And a case of 0 responses if that is legal.

Zero response is a legal case for decoupled model so I think we should treat it legal case for LLM model as well.

Reply:
The (hopefully) last thing I'd like to see is a blend. Specifically a mixture of a request with no response and a request with multiple responses. I want to ensure that the "no response" code doesn't trigger unless ALL requests have no responses.

Reply:
Added a test case with mixture of responses